### ISSUE TITLE
Issue with installation 

### ISSUE THREAD

**Issue by hqs811** [mathworks/matlab-engine-for-python#19]
Hi,

when I try 

python -m pip install matlabengine, I had the following :

Does anyone have an idea of how to fix it? Thanks! 

  × Building wheel for matlabengine (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [50 lines of output]
      running bdist_wheel
      running build
      running build_py
      Traceback (most recent call last):
.....
        File "<string>", line 379, in run
        File "<string>", line 215, in get_matlab_root_from_windows_reg
        File "<string>", line 249, in _find_matlab_key_from_windows_registry
        File "<string>", line 279, in _check_matlab_ver_against_engine
      TypeError: _get_engine_ver_major_minor() missing 1 required positional argument: 'id'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for matlabengine
Failed to build matlabengine
ERROR: Could not build wheels for matlabengine, which is required to install pyproject.toml-based projects

**Comment by YZ-Yan** [comment]
Did you solve the problem?

**Comment by Ll0veU** [comment]
I failed, too. 
When I try 

```
python -m pip install matlabengine
```

The output is as follows:

```
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting matlabengine
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/bf/6c/15b88db41c607055fb0058dd1737084c3843e810580230061b3a6c24339f/matlabengine-23.2.3.tar.gz (18 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: matlabengine
  Building wheel for matlabengine (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for matlabengine (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [54 lines of output]
      running bdist_wheel
      running build
      running build_py
      _create_path_list returned: ['', '/Applications/MATLAB_R2023a.app/bin/maci64', '/Applications/MATLAB_R2023a.app/bin/maci64', '/Applications/MATLAB_R2023a.app/bin/maci64']
      search_path_for_directory_unix returned:
      Traceback (most recent call last):
        File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 410, in build_wheel
          return self._build_with_temp_dir(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 395, in _build_with_temp_dir
          self.run_setup()
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 415, in <module>
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/wheel/bdist_wheel.py", line 368, in run
          self.run_command("build")
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/cj/fzt_j_4552q33_j8bk_hkvcc0000gn/T/pip-build-env-9jucmvgx/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 405, in run
      RuntimeError: MATLAB R2023b installation not found. Install to default location, or add <matlabroot>/bin/maca64 to DYLD_LIBRARY_PATH, where <matlabroot> is the root of a MATLAB R2023b installation.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for matlabengine
Failed to build matlabengine
ERROR: Could not build wheels for matlabengine, which is required to install pyproject.toml-based projects
```

By the way, OS: macOS 14.0 Sonoma; Py Env: Python 3.11.2, pip 24.0 from /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pip (python 3.11).

The command was execute in directory `/Applications/MATLAB_R2023a.app/bin/maci64/`.

**Comment by alanfmw** [comment]
Your error mentions **R2023b**:

_RuntimeError: MATLAB **R2023b** installation not found. Install to default location, or add <matlabroot>/bin/maca64 to DYLD_LIBRARY_PATH, where <matlabroot> is the root of a MATLAB **R2023b** installation._

However, you say that you have **R2023a** installed:

_The command was execute in directory /Applications/MATLAB_**R2023a**.app/bin/maci64/._

This means that you should be executing 

`python -m pip install matlabengine==9.14`

since 9.14 corresponds to R2023a. If you leave off the "==9.14", it will try to install against the latest supported release, which is currently R2023b.

The problem mentioned in the first post in the thread comes from a different source, and was fixed a while ago.

**Comment by Ll0veU** [comment]
Thanks, the problem has been solved~



---- Replied Message ----
| From | ***@***.***> |
| Date | 03/20/2024 23:20 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [mathworks/matlab-engine-for-python] Issue with installation (Issue #19) |

Closed #19 as completed.
